music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -0,0 +1,437 @@
+                label="",
+            )
+            yield BrowseFolder(
+                item_id="favorites",
+                provider=self.domain,
+                path=path + "favorites",
+                name="Favorites",
+                label="",
+            )
+            yield BrowseFolder(
+                item_id="categories",
+                provider=self.domain,
+                path=path + "categories",
+                name="Categories",
+                label="",
+            )
+            yield BrowseFolder(
+                item_id="decades",
+                provider=self.domain,
+                path=path + "decades",
+                name="Decades",
+                label="",
+            )
+
+        if sub_path == "channels":
+            for channel_id, channel in self._channels_by_id.items():
+                yield self._parse_radio(channel)
+
+        if sub_path == "favorites":
+            for channel in self._favorite_channels:
+                yield self._parse_radio(channel)
+
+        if sub_path == "decades":
+            for decade_range in self._channels_by_decade.keys():
+                decade = str(decade_range[0])
+                decade_name = f"{decade[-2:]}'s"
+                decade_name = f"{decade_name} and 60's" if decade_name == "50's" else decade_name
+                decade_name = f"20{decade_name}" if decade_name in ["00's", "10's", "20's"] else decade_name
+                yield BrowseFolder(
+                    item_id=decade,
+                    provider=self.domain,
+                    path=path + "/" + decade,
+                    name=decade_name,
+                    label="",
+                )
+
+        if sub_path.startswith("decades/"):
+            year = int(sub_sub_path)
+            channels = self._get_channels_for_year(year)
+            for channel in channels:
+                yield self._parse_radio(channel)
+
+        if sub_path == "categories":
+            for category_key, category in self._categories.items():
+                if not category.name.endswith('Decade') and not category.name.endswith('Decades'):
+                    yield BrowseFolder(
+                        item_id=category_key,
+                        provider=self.domain,
+                        path=path + "/" + category_key,
+                        name=category.name,
+                        label="",
+                    )
+
+        if sub_path.startswith("categories/"):
+            category_key = sub_sub_path
+            channels = self._channels_by_category[category_key]
+            for channel in channels:
+                yield self._parse_radio(channel)
+
+    async def get_library_radios(self) -> AsyncGenerator[Radio, None]:
+        """Retrieve library/subscribed playlists from the provider."""
+        for channel_guid, channel in self._channels_by_id.items():
+            yield self._parse_radio(channel)
+
+    async def get_radio(self, channel_id: str) -> Union[Radio, None]:

```

Comment:
you may use new style type annotations..
so instead of Union(Radio, None] you can just use Radio | None